src/dmtcpplugin.cpp

```diff
@@ -361,23 +361,32 @@ dmtcp_get_restart_env(const char *name,   // IN
 
   DmtcpGetRestartEnvErr_t rc = RESTART_ENV_NOTFOUND;
 
-  char env_buf[RESTART_ENV_MAXSIZE] = { 0 }; // All "name=val" strings must be
-                                             // shorter than this.
-
   if (name == NULL || value == NULL) {
     close(env_fd);
     return RESTART_ENV_NULL_PTR;
   }
 
+  struct stat statbuf;
+  size_t size = RESTART_ENV_MAXSIZE;
+  char *env_buf = NULL;
+
+  if (fstat(env_fd, &statbuf) == 0) {
+    if (statbuf.st_size > RESTART_ENV_MAXSIZE) {
```

Comment:
Why even bother about `RESTART_ENV_MAXSIZE` here? We can just kill it entirely and continue with the size computation.